client/jetpack-connect/controller.js

```diff
@@ -74,12 +78,19 @@ const analyticsPageTitleByType = {
 };
 
 export function offerResetContext( context, next ) {
+	debug( 'controller: offerResetContext', context.params );
 	context.header = <StoreHeader />;
 	context.footer = <StoreFooter />;
 	next();
 }
 
 const getPlanSlugFromFlowType = ( type, interval = 'yearly' ) => {
+	// Return early if `type` is already a real product slug that is part
+	// of the Offer Reset flow.
+	if ( OFFER_RESET_FLOW_TYPES.includes( type ) ) {
+		return type;
+	}
+
```

Comment:
Currently, the Connect Flow uses types such as 'pro', 'jetpack_search', 'backup', 'realtimebackup', 'scan', to name a few, combined with an internal such as 'yearly' or 'monthly'. With these two variables, the code construct the final product slug. 

Offer Reset doesn't requires this, since the CTA in product cards already have the final product slug in them. In short, we need to support all product slugs supported by Offer Reset as well as the old ones.